src/utils/useFormAndValidate.js

```diff
@@ -0,0 +1,39 @@
+import { useState, useCallback } from "react";
+
+export const useFormAndValidate = () => {
```

Comment:
You should put it in a separate file  `useFormAndValidation.js` in folder `hooks` and then import it to the component that needs validation. It's common to keep customs hooks in folder `hooks`